login/service.go

```diff
@@ -137,16 +137,16 @@ func (keycloak *KeycloakOAuthProvider) Login(ctx *app.LoginLoginContext, config
 
 	// First time access, redirect to oauth provider
 
-	redirectURL, err := keycloak.beforeRedirectToLogin(ctx, ctx.Redirect, ctx.APIClient, ctx.RequestData, config, serviceConfig)
+	redirectURL, err := keycloak.saveParamsAndReferrer(ctx, ctx.Redirect, ctx.APIClient, ctx.RequestData, config, serviceConfig)
 	if err != nil {
 		return jsonapi.JSONErrorResponse(ctx, err)
 	}
 	ctx.ResponseData.Header().Set("Location", *redirectURL)
 	return ctx.TemporaryRedirect()
 }
 
-// beforeRedirectToLogin saves referer and api_client parameter
-func (keycloak *KeycloakOAuthProvider) beforeRedirectToLogin(ctx context.Context, redirect *string, apiClient *string, request *goa.RequestData, config oauth.OauthConfig, serviceConfig LoginServiceConfiguration) (*string, error) {
+// saveParamsAndReferrer saves referer and api_client parameter
+func (keycloak *KeycloakOAuthProvider) saveParamsAndReferrer(ctx context.Context, redirect *string, apiClient *string, request *goa.RequestData, config oauth.OauthConfig, serviceConfig LoginServiceConfiguration) (*string, error) {
```

Comment:
I'm sorry. I didn't actually look at this method closely before. saveParamsAndReferrer is not accurate name for this method. This method saves the param & referrer but also returns the keycloak redirect URL.

This method actually should be the AuthCodeURL() method. It's the same method. saveParamsAndReferrer() == AuthCodeURL(). So just rename it to AuthCodeURL().